cudamapper/src/cudamapper_utils.hpp

```diff
@@ -49,5 +49,31 @@ namespace cudamapper
 /// \param unfused_overlaps vector of overlaps, sorted by (query_id, target_id) combination and query_start_position
 void fuse_overlaps(std::vector<Overlap>& fused_overlaps, const std::vector<Overlap>& unfused_overlaps);
 
+std::string string_slice(const std::string& s, std::size_t start, std::size_t end);
+
+/// \brief Given a string s, produce its kmers (length <kmer-length>) and return them as a vector of strings.
+/// \param s A string sequence to kmerize.
+/// \param kmer_length A kmer length to use for producing kmers.
+/// \param stride The number of bases to skip when selecting kmers (most often, this should be equal to 1).
+std::vector<std::string> kmerize_string(const std::string& s, int kmer_length, int stride);
+
+/// \brief Given two sorted vectors of comparable types, return a size_t count of the number of shared elements.
```

Comment:
Let's say that vector `a` has `m` elements with value `x` and vector `b` `n` such elements. Should this results in `1` or `m*n` shared elements? I would assume former but your implementation would actually return `min(m, n)`

Reply:
I intended for this to return `min(m,n)`; I don't want to discount duplicates. We could add a version to return *unique* shared elements in a new PR.

Reply:
I'm fine with whatever version suits your algorithm, no need to add function we don't need at the moment.
I would just extend the comment so it's clear to the user without looking into implementation.

Reply:
Ah, got it! This is now documented in 069cae0.